ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,898 @@
+/* *
+ *
+ *  Highcharts Breadcrumbs module
+ *
+ *  Authors: Grzegorz Blachliński, Karol Kołodziej
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+import type {
+    AlignValue,
+    VerticalAlignValue
+} from '../Core/Renderer/AlignObject';
+import type {
+    CSSObject
+} from '../Core/Renderer/CSSObject';
+import type SVGElement from '../Core/Renderer/SVG/SVGElement';
+import Chart from '../Core/Chart/Chart.js';
+import H from '../Core/Globals.js';
+import O from '../Core/Options.js';
+const {
+    defaultOptions
+} = O;
+import U from '../Core/Utilities.js';
+const {
+    addEvent,
+    defined,
+    extend,
+    format,
+    merge
+} = U;
+
+/**
+ * Internal types
+ * @private
+ */
+declare global {
+    namespace Highcharts {
+        interface BreadcrumbsOptions {
+            buttonPadding?: number;
+            enabled?: boolean;
+            events?: BreadcrumbsButtonsEventsOptions;
+            floating?: boolean;
+            format?: string;
+            formatter?: BreadcrumbsButtonsFormatter;
+            position?: BreadcrumbsAlignOptions;
+            separator?: string;
+            showOnlyLast?: boolean;
+            style?: CSSObject;
+            useHTML?: boolean;
+            zIndex?: number;
+        }
+        interface BreadcrumbsAlignOptions {
+            align?: AlignValue;
+            verticalAlign?: VerticalAlignValue;
+            alignByTranslate?: boolean;
+            x: number;
+            y: number;
+        }
+        interface BreadcrumbsClickCallbackFunction {
+            (e: Event, breadcrumbs: Breadcrumbs): (boolean|undefined);
+        }
+        interface BreadcrumbsButtonsEventsOptions {
+            click?: BreadcrumbsClickCallbackFunction;
+        }
+        interface BreadcrumbsButtonsFormatter {
+            (
+                breadcrumb: Array<any>,
+                breadcrumbs: Breadcrumbs
+            ): (string);
+        }
+        interface Options {
+            breadcrumbs?: BreadcrumbsOptions;
```

Comment:
Breadcrumbs is a navigation aid feature. Perhaps we should place options in `navigation.breadcrumbs`?

Reply:
We use navigation also in stockTools, I'm afraid that it can be tricky here.

Reply:
`drilldown.breadcrumbs` is unfortunate because in `treemap` we don't use that and creating that whole structure might be confusing for the user. 
`navigation.breadcrumbs` sound reasonable and maybe more natural in this case where breadcrumbs are used to navigate through the chart.
What do you think @TorsteinHonsi ?

Reply:
I think we should have `drilldown.breadcrumbs: BreadCrumbsOptions` and `plotOptions.treemap.breadcrumbs: BreadCrumbsOptions`. Sunburst will inherit from treemap. 

The idea is that this places the drill-up button for drilldown and the traverse-up button for treemap.

Reply:
@TorsteinHonsi We have changed that- the breadcrumbs options now are available inside `drilldown.breadcrumbs` and `treemap.breadcrumbs`.
What I would like to ask is where the breadcrumbs class should be?  Now you might reach it in the `chart`. Should we move it to `drilldown` and `treemap`, or can we leave that as it is now on the `chart`?

Reply:
Not sure if I understand. Do we need to register the class anywhere at all? Can't we just import the module into the drilldown and the treemap modules?